### ISSUE TITLE
Update TSNE docs to clarify what is passed in on instantiation or fit

### ISSUE THREAD

**Issue by rebeccabilbro** [DistrictDataLabs/yellowbrick#684]
**Describe the bug**
Currently, the TSNE docs [show an example](http://www.scikit-yb.org/en/latest/api/text/tsne.html#t-sne-corpus-visualization) where a variable named "labels" is passed into the `fit` method; however this variable actually stores `y` (the target), not the labels (which the `fit` method can derive from `y`). This has [lead to some confusion](https://github.com/DistrictDataLabs/yellowbrick/pull/659) about what values should and can be passed in when and where.

**Proposed Fix**
Update the TSNE docs to something like:

```python
from yellowbrick.text import TSNEVisualizer
from sklearn.feature_extraction.text import TfidfVectorizer

# Load the data and create document vectors
corpus = load_corpus('hobbies')
tfidf  = TfidfVectorizer()

X = tfidf.fit_transform(corpus.data)
y = corpus.target

tsne = TSNEVisualizer()
tsne.fit(X, y)
tsne.poof()
```

We should also add another example to illustrate how a user can pass in the labels as a list on instantiation, e.g.:

```python
...

labels = corpus.categories

tsne = TSNEVisualizer(labels=labels)
tsne.fit(X, y)
tsne.poof()
```

Sincere thanks to @jeromemassot for alerting us to this point of confusion!

**Comment by 53X** [comment]
Can I take this issue?

**Comment by bbengfort** [comment]
@53X sure that would be great, thanks.

**Comment by Naba7** [comment]
@rebeccabilbro I have created a PR regarding this issue
